drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTableModellerViewImpl.java

```diff
@@ -0,0 +1,1042 @@
+                                            return;
+                                        }
+                                        String cm = GuidedDecisionTableConstants.INSTANCE.DeleteConditionColumnWarning0( c.getHeader() );
+                                        if ( Window.confirm( cm ) ) {
+                                            presenter.getActiveDecisionTable().deleteColumn( c );
+                                        }
+                                    }
+                                } );
+    }
+
+    private Widget getActionsWidget() {
+        actionsConfigWidget = new VerticalPanel();
+        return actionsConfigWidget;
+    }
+
+    @Override
+    public void refreshActionsWidget( final List<ActionCol52> actionColumns ) {
+        this.actionsConfigWidget.clear();
+
+        if ( actionColumns == null || actionColumns.size() == 0 ) {
+            disclosurePanelActions.setOpen( false );
+            return;
+        }
+
+        //Each Action is a row in a vertical panel
+        final VerticalPanel actionsPanel = new VerticalPanel();
+        this.actionsConfigWidget.add( actionsPanel );
+
+        //Add Actions to panel
+        final boolean isEditable = presenter.isActiveDecisionTableEditable();
+        for ( ActionCol52 actionColumn : actionColumns ) {
+            HorizontalPanel hp = new HorizontalPanel();
+            if ( isEditable ) {
+                hp.add( removeAction( actionColumn ) );
+            }
+            hp.add( editAction( actionColumn ) );
+            Label actionLabel = makeColumnLabel( actionColumn );
+            hp.add( actionLabel );
+            actionsPanel.add( hp );
+        }
+    }
+
+    private SmallLabel makeColumnLabel( final ActionCol52 actionColumn ) {
+        SmallLabel label = new SmallLabel( actionColumn.getHeader() );
+        if ( actionColumn.isHideColumn() ) {
+            label.setStylePrimaryName( GuidedDecisionTableResources.INSTANCE.css().columnLabelHidden() );
+        }
+        return label;
+    }
+
+    private Widget editAction( final ActionCol52 actionColumn ) {
+        Image edit = GuidedDecisionTableImageResources508.INSTANCE.Edit();
+        edit.setAltText( GuidedDecisionTableConstants.INSTANCE.EditThisActionColumnConfiguration() );
+        return new ImageButton( edit,
+                                GuidedDecisionTableConstants.INSTANCE.EditThisActionColumnConfiguration(),
+                                new ClickHandler() {
+                                    public void onClick( ClickEvent w ) {
+                                        presenter.getActiveDecisionTable().editAction( actionColumn );
+                                    }
+                                } );
+    }
+
+    private Widget removeAction( final ActionCol52 actionColumn ) {
+        if ( actionColumn instanceof LimitedEntryBRLActionColumn ) {
```

Comment:
These look eerily similar to the `edit|removeCondition(..)` above - if it is possible to reuse the code without too much bother, that would be great. And at least the `image` in the `removeAction(..)` could be set  up at the beginning of the method for all column types.